test/integration/scheduler_perf/config/performance-config.yaml

```
@@ -1383,3 +1383,32 @@
     params:
       gatedPods: 10000
       measurePods: 20000
+
+# This test case simulates the scheduling when pods selected to schedule have deletionTimestamp set.
+# There was a memory leak related to this path of code fixed in:
+# https://github.com/kubernetes/kubernetes/pull/126962
+# Main goal of this test case is to verify if InFlightEvents is empty after the test.
+- name: SchedulingDeletedPodsWithFinalizers
+  featureGates:
+    SchedulerQueueingHints: true
+  defaultPodTemplatePath: config/templates/light-pod.yaml
+  workloadTemplate:
+  - opcode: createNodes
+    count: 1000
+  - opcode: createPods
+    # Create pods with finalizers and delete them before they are scheduled.
+    # DeletionTimestamp field should be populated then, 
+    # but a few pods should still be chosen into the scheduling.
+    countParam: $deletingPods
+    podTemplatePath: config/templates/pod-with-finalizer.yaml
+    skipWaitToCompletion: true
+    deletePodsPerSecond: 1000
```

```suggestion
    deletePodsPerSecond: 100
```

Would this have made the test more likely to fail?

---

Right, that leaves more InFlightEvents not cleaned.